cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+    NodeInstanceType,
+    nodeInstanceTypeValidation,
+    NodeProfile,
+} from '../../../../shared/components/FormInputComponents/NodeProfile/NodeProfile';
+import { StepProps } from '../../../../shared/components/wizard/Wizard';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+// NOTE: icons must be imported
+const nodeInstanceTypes: NodeInstanceType[] = [
+    {
+        id: 'single-node',
+        label: 'Single node',
+        icon: 'block',
+        description: 'Create a single control plane node with a Standard_D2s_v3 instance type',
+    },
+    {
+        id: 'high-availability',
+        label: 'High availability',
+        icon: 'blocks-group',
+        description: 'Create a multi-node control plane with a Standard_D2s_v3 instance type',
+    },
+    {
+        id: 'compute-optimized',
+        label: 'Production-ready (High availability)',
+        icon: 'blocks-group',
+        isSolidIcon: true,
+        description: 'Create a multi-node control plane with a Standard_D4s_v3 instance type',
+    },
+];
+
+type AZURE_CLUSTER_SETTING_STEP_FIELDS = AZURE_FIELDS.CLUSTER_NAME | AZURE_FIELDS.INSTANCE_TYPE;
+
+interface AzureClusterSettingFormInputs {
+    [AZURE_FIELDS.CLUSTER_NAME]: string;
+    [AZURE_FIELDS.INSTANCE_TYPE]: string;
+}
+
+export function AzureClusterSettingsStep(props: Partial<StepProps>) {
+    const { currentStep, deploy, handleValueChange } = props;
+    const { azureState, azureDispatch } = useContext(AzureStore);
+    const azureClusterSettingsFormSchema = yup.object(createYupSchemaObject()).required();
+    const methods = useForm<AzureClusterSettingFormInputs>({
+        resolver: yupResolver(azureClusterSettingsFormSchema),
+    });
+
+    const {
+        handleSubmit,
+        formState: { errors },
+        register,
+        setValue,
+    } = methods;
+
+    let initialSelectedInstanceTypeId = azureState[AZURE_FIELDS.INSTANCE_TYPE];
+    if (!initialSelectedInstanceTypeId) {
+        initialSelectedInstanceTypeId = nodeInstanceTypes[0].id;
+        setValue(AZURE_FIELDS.INSTANCE_TYPE, initialSelectedInstanceTypeId);
+    }
+    const [selectedInstanceTypeId, setSelectedInstanceTypeId] = useState(initialSelectedInstanceTypeId);
+
+    const canContinue = (): boolean => {
+        return Object.keys(errors).length === 0;
+    };
+
+    // TODO: just deactivate button until no errors
+    const onSubmit: SubmitHandler<AzureClusterSettingFormInputs> = (data) => {
+        if (canContinue() && deploy) {
+            deploy();
+        }
+    };
+
+    const onFieldChange = (data: string, field: AZURE_CLUSTER_SETTING_STEP_FIELDS) => {
+        if (handleValueChange) {
+            handleValueChange(INPUT_CHANGE, field, data, currentStep, errors);
+            setValue(field, data, { shouldValidate: true });
+        }
+    };
+
+    const onClusterNameChange = (clusterName: string) => {
+        onFieldChange(clusterName, AZURE_FIELDS.CLUSTER_NAME);
```

Comment:
should we be dispatching these values to the store here?

Reply:
@swalner-vmware I think onFieldChange function can dispatch the event for the value change.

Reply:
I like it, Raymond

Reply:
It looks like this is being done through Form.reducer.tsx. We pass the azureState and azureDispatch into the wizard, so it looks like setValue is dispatching the action to the correct store but I do not see how that connection is made yet. Will ask Raymond to get us up to speed.